src/test_fixture/sharing.rs

```diff
@@ -18,6 +18,24 @@ pub fn share<F: Field, R: RngCore>(input: F, rng: &mut R) -> [Replicated<F>; 3]
     ]
 }
 
+/// Shares `input` into 3 maliciously secure replicated secret shares using the provided `rng` implementation
+///
+#[allow(clippy::missing_panics_doc)]
+pub fn share_malicious<F: Field, R: RngCore>(
+    input: F,
+    r: [Replicated<F>; 3],
+    rng: &mut R,
+) -> [MaliciousReplicated<F>; 3] {
+    share(input, rng)
+        .iter()
+        .enumerate()
+        .map(|(i, share)| MaliciousReplicated::new(*share, r[i]))
```

Comment:
I don’t think this is right. The second argument should be a sharing of input times r. This appears be be just r.

Reply:
I think that you want:

```rust
let r = validate_and_reconstruct((r[0], r[1], r[2]));
let rx = r * input;
zip(share(input, rng), share(rx, rng))
  .map(|(x, rx)| MaliciousReplicated::new(*x, *rx))
  .collect::<Vec<_>>()
  .try_into()
  .unwrap()
```

Reply:
yep that was a bug. I decided to generate `r` inside `share_malicious` given that I have access to PRNG there. Lmk if that is not ergonomic (we could always reconstruct $r$ from $[rx]$)